src/array.ts

```diff
@@ -144,6 +145,16 @@ Array.prototype.unique = function () {
   return [...new Set(this)];
 };
 
+/** Would be cool to allow an array of keys to make the criteria of "unique" more flexible */
+Array.prototype.uniqueObjectsByKey = function <T>(key: string) {
```

Comment:
Wonder if `function <T extends object>` matters?

Reply:
I played around with this a bit and it didn't seem to provide better type hints.